internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -409,17 +411,42 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 		log.WithError(err).Error("failed to get corresponding infraEnv")
 		return false, err
 	}
+
+	// 1 - Check if agent image image is overriden
 	ironicAgentImage, ok := infraEnv.GetAnnotations()[ironicAgentImageOverrideAnnotation]
-	if !ok || ironicAgentImage == "" {
+	if ok && ironicAgentImage != "" {
+		log.Infof("Using override ironic agent image (%s) for infraEnv %s", ironicAgentImage, infraEnv.Name)
+	}
+
+	// 2 - Check if ICC config is available, use agent image if CPU architecture is supported.
+	var iccConfig *ICCConfig
+	if ironicAgentImage == "" {
+		iccConfig, err = r.BMOUtils.GetICCConfig()
+		if err != nil {
+			log.WithError(err).Info("ICC configuration is not available, will continue without it")
+		}
+	}
+	if iccConfig != nil {
+		var architectures []string
+		architectures, err = r.OcRelease.GetReleaseArchitecture(log, iccConfig.IronicAgentImage, r.ReleaseImageMirror, infraEnvInternal.PullSecret)
```

Comment:
Does this work? I'm surprised this isn't specific to release images.

Reply:
I did not get any issues in my cluster so far. I'll double check.

Reply:
The [code](https://github.com/openshift/assisted-service/blob/master/internal/oc/release.go#L249) basically does a `oc image info --output json quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e3813103687092e9016edf7d40ea7394d8cddbc773a750f5a67317f751d3eee7 --registry-config ~/pull-secret | jq .config.architecture` (it's a bit more complicated if the image is multi-arch as is uses skopeo).

I didn't realize that `Release` in function name meant "OCP release", but the code behind it looks quite generic.

Reply:
Yeah, that was supposed to be for release images ... I mean I suppose this is an image in the release so it _could_ make sense. I just worry that in the future someone might change this function so that it does only work for release images and break this use.

What do you think about creating a new function that just calls the old one called `GetImageArchitecture`? Maybe we can then fully move over to the new name in a followup?

Reply:
I agree, I thought about the same thing this morning.